lib/home/schedule/src/schedule_editor.dart

```diff
@@ -1,14 +1,309 @@
+import 'package:intl/intl.dart';
+import 'package:table_calendar/table_calendar.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:thc/home/users/src/all_users.dart';
 import 'package:thc/the_good_stuff.dart';
-import 'package:thc/utils/widgets/placeholders.dart';
 
-class ScheduleEditor extends StatelessWidget {
+class ScheduleEditor extends StatefulWidget {
   const ScheduleEditor({super.key});
 
+  @override
+  State<ScheduleEditor> createState() => _ScheduleEditorState();
+}
+
+class _ScheduleEditorState extends State<ScheduleEditor> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final TextEditingController _event = TextEditingController();
+  final TextEditingController _startDate = TextEditingController();
+  final TextEditingController _endDate = TextEditingController();
+  final TextEditingController _startTime = TextEditingController();
+  final TextEditingController _endTime = TextEditingController();
+
+  Director? director;
+
+  DateTime _selectedDay = DateTime.now();
+  Map<DateTime, List<dynamic>> _events = {};
+  CalendarFormat _calendarFormat = CalendarFormat.month;
+  DateTime _focusedDay = DateTime.now();
+
+  final eventsCollection = FirebaseFirestore.instance.collection('events');
+
+  @override
+  void initState() {
+    super.initState();
+    fetchEvents();
+  }
+
+  void submit() {
+    if (_formKey.currentState!.validate()) {
+      addEvent();
+      navigator.snackbarMessage('${_event.text} event added!');
+    }
+  }
+
+  void addEvent() {
+    final newEvent = {
+      'title': _event.text,
+      'startDate': _startDate.text,
+      'endDate': _endDate.text,
+      'startTime': _startTime.text,
+      'endTime': _endTime.text,
+      'directorId': director?.firestoreId ?? '',
+    };
```

Comment:
Here's an example of the JSON schema we should match:

https://github.com/theheartcenter-inc/thc/blob/513c5943da2b18720da86c9b54c2039608c29e50/lib/home/schedule/schedule.dart#L17-L24

Reply:
got it! Thanks Nate 😄

Reply:
It looks like the name of the collection was updated (which is great!) but not the JSON schema here…